### PR TITLE
New workflow to add CodeQL PRs to projects

### PR DESCRIPTION
**Pull request by sidshank**
This workflow introduces some automation to add issues that are tagged with ANY of the labels:
- `C#`
- `Go`
- `Java`
- `Kotlin`

To be added to the appropriate GitHub project.

The workflow is generically named "Add CodeQL pull requests to project" with the intent that future jobs might be added to the workflow to triage PRs in this repo to other projects.

### CODE REVIEW

**Review by krukow — APPROVED** [review]

**Comment by sidshank** [comment]
@github/codeql-ci-reviewers 👋🏼 I know I can go ahead and merge this in, but since you're tagged as CODEOWNERS for this folder in the repo, I want to check in with you folks before I push these changes in. Please let me know if you have any concerns. 🙇🏼

File: .github/workflows/add-codeql-prs-to-project.yml
Comment (criemen) [inline .github/workflows/add-codeql-prs-to-project.yml:17]: As this is a first-party action, you can pull it in via a tag or branch - it is only 3rd-party actions where we need to fix the commit SHA

**Review by criemen — COMMENTED** [review]
The PR looks fine to me (one minor nitpick).

However, I know that @adityasharad often has concerns around CI jobs that run on public repositories that interact with internal resources. If you wouldn't mind too much, could you check in with him when he's back from vacation about this job?

**Comment by sidshank** [comment]
> I know that @adityasharad often has concerns around CI jobs that run on public repositories that interact with internal resources. If you wouldn't mind too much, could you check in with him when he's back from vacation about this job?

I don't mind at all! This isn't urgent, so I'm happy to consult with him before I merge this in. ✌🏼 Thanks for your review @criemen.

File: .github/workflows/add-codeql-prs-to-project.yml
Comment (adityasharad) [inline .github/workflows/add-codeql-prs-to-project.yml:13]: Try this. The condition here should ensure the entire job only runs when the PR has a relevant label. This keeps the compute usage low and avoids cluttering the Actions tab for this repo.
See https://docs.github.com/en/actions/learn-github-actions/expressions#example-using-an-object-filter for docs on this syntax. This is a bit clunky, because there is no subset operator in the expression language and we have to list all the possible labels.
```suggestion
    runs-on: ubuntu-latest
    if: |
      contains(github.event.pull_request.labels.*.name, 'C#') ||
      contains(github.event.pull_request.labels.*.name, 'Go') ||
      contains(github.event.pull_request.labels.*.name, 'Java') ||
      contains(github.event.pull_request.labels.*.name, 'Kotlin')
    permissions:
```

File: .github/workflows/add-codeql-prs-to-project.yml
Comment (adityasharad) [inline .github/workflows/add-codeql-prs-to-project.yml:14]: I wonder if you need `write` here. Let's try and see.

**Review by adityasharad — COMMENTED** [review]
Looks reasonable. As discussed, try this in another public repo and let's see if the `repository-projects` permissions does what we expect.

**Comment by sidshank** [comment]
Closing in favour of an alternate approach.

**Pull request closed without merging**
